protocols/1540-single-cell-discoveries/ngs_prep.ot2.py

```diff
@@ -0,0 +1,179 @@
+from opentrons import labware, instruments, modules, robot
+
+metadata = {
+    'protocolName': 'NGS Prep',
+    'author': 'Nick <protocols@opentrons.com>',
+    'source': 'Custom Protocol Request'
+}
+
+# labware
+sample_plate = labware.load('opentrons-aluminum-block-96-PCR-plate', '2')
+new_plate = labware.load('opentrons-aluminum-block-96-PCR-plate', '3')
+trough = labware.load('trough-12row', '4')
+tips10_rack = [labware.load('tiprack-10ul', slot)
+               for slot in ['5', '6', '7']]
+tips300_rack = [labware.load('tiprack-200ul', slot)
+                for slot in ['8', '9', '10', '11']]
+
+# modules
+magdeck = modules.load('magdeck', '1')
+mag_plate = labware.load('opentrons-aluminum-block-96-PCR-plate',
+                         '1',
+                         share=True)
+
+# pipettes
+p10 = instruments.P10_Single(
+    mount='right',
+    tip_racks=tips10_rack
+)
+
+p300 = instruments.P300_Single(
+    mount='left',
+    tip_racks=tips300_rack
+)
+
+# reagent setup
+beads = trough.wells('A1')
+EtOH = trough.wells('A2')
+water = trough.wells('A3')
+liquid_trash = trough.wells('A4', to='A12')
+
+
+def run_custom_protocol(number_of_columns: int = 12,
+                        start_column: int = 1,
+                        volume_of_beads: float = 10.0,
+                        volume_of_first_supernatant_to_remove: float = 50,
+                        number_of_washes: int = 2,
+                        minutes_to_dry: int = 10,
+                        resuspension_volume: float = 100,
+                        volume_of_second_supernatant_to_remove: float = 50):
+    # check for valid columns
+    if number_of_columns + start_column > 13:
+        raise Exception('Invalid columns.')
+
+    # check for valid number of number of washes
+    if number_of_washes > 9:
+        raise Exception('Too many EtOH washes.')
+
+    # setup sample locations
+    cols = sample_plate.columns(str(start_column), length=number_of_columns)
+    samples = [well for col in cols for well in col]
+
+    mag_cols = mag_plate.columns(str(start_column), length=number_of_columns)
+    mag_samples = [well for col in mag_cols for well in col]
+
+    super_cols = new_plate.columns(str(start_column), length=number_of_columns)
+    super_samples = [well for col in super_cols for well in col]
+
+    # mix and transfer beads
+    p300.pick_up_tip()
+    p300.mix(10, 300, beads)
+    p300.drop_tip()
+
+    if volume_of_beads > 10:
+        pipette = p300
+    else:
+        pipette = p10
+    pipette.transfer(volume_of_beads,
+                     beads,
+                     [s.top() for s in samples],
+                     mix_after=(10, volume_of_beads))
+
+    # incubate at room temperature for 15 minutes
+    pipette.delay(minutes=15)
```

Comment:
let's assume the plate is already on the magdeck, so no need to pause and move plate